Sources/PT.PM.Common/TextUtils.cs

```diff
@@ -39,22 +34,22 @@ public static int FirstIndexOf(this string str, int index, bool whitespace)
             return i;
         }
 
-        public static TextSpan Union(this IEnumerable<TextSpan> textSpans)
+        public static TextSpan Union(this IList<TextSpan> textSpans)
         {
-            if (textSpans == null || textSpans.Count() == 0)
+            if (textSpans == null || textSpans.Count == 0)
             {
                 return TextSpan.Zero;
             }
 
-            var resultTextSpan = textSpans.First();
-            if (textSpans.Count() == 1)
+            var resultTextSpan = textSpans[0];
+            if (textSpans.Count == 1)
             {
                 return resultTextSpan;
             }
 
-            foreach (TextSpan textSpan in textSpans.Skip(1))
+            for (int i = 1; i < textSpans.Count; i++)
```

Comment:
I think `foreach` version conveyed intent clearer, it's easier to miss the fact that indexing starts with 1 than `Skip(1)` call.

In theory, `foreach` version could produce slightly faster code. Anyway, if actual implementation of both interfaces is the same there would be no significant performance difference, so I don't see the point of making the interface more concrete. What was the reasoning behind the decision?

Reply:
Rider highlighted the warning of `Potential multiple enumerations`. To get rid, the input `IEnumerable` should be transformed to list or array. That's why I decided to replace it with a more concrete interface. Moreover, `IEnumerable` produces overhead code in general.